controllers/warehouseController.js

```diff
@@ -0,0 +1,58 @@
+const Joi = require("joi");
+const knex = require("knex")(require("../knexfile"));
+
+const updateWarehouseDetails = async (req, res) => {
+  try {
+    const warehouseId = parseInt(req.params.id);
+
+    if (!warehouseId || isNaN(warehouseId)) {
+      return res.status(400).json({ message: "Invalid warehouse ID." });
+    }
+
+    const warehouseSchema = Joi.object({
+      warehouse_name: Joi.string().required(),
+      address: Joi.string().required(),
+      city: Joi.string().required(),
+      country: Joi.string().required(),
+      contact_name: Joi.string().required(),
+      contact_position: Joi.string().required(),
+      contact_phone: Joi.string()
+        .pattern(/^\+?[0-9\s-().]*$/)
+        .required(),
+      contact_email: Joi.string().email().required(),
+    });
+
+    const { error } = warehouseSchema.validate(req.body, { abortEarly: false });
+    if (error) {
+      return res.status(400).json({
+        message: error.details.map((detail) => detail.message).join(", "),
+      });
+    }
+
+    const warehouseExists = await knex("warehouses")
+      .where("id", warehouseId)
+      .first();
+    if (!warehouseExists) {
+      return res.status(404).json({ message: "Warehouse not found." });
+    }
+
+    await knex("warehouses").where("id", warehouseId).update(req.body);
+
+    const updatedWarehouse = await knex("warehouses")
+      .where("id", warehouseId)
+      .first();
+
```

Comment:
You are retrieving the updated warehouse after executing the update query, but there is a chance the update operation could fail, and you are not checking whether it was successful. I recommend adding a check to ensure the update was successful before querying the database for the updated information. If the update fails, you can return an appropriate error response instead of making an additional DB call.

```

const updateCount   = await knex("warehouses").where("id", warehouseId).update(req.body);

// 0 ==  failed
//1
if (updateCount === 0) {
  return res.status(500).json({ message: "Update failed: Warehouse not found or no changes made. Try again later " });
}
```

**NOTE** 
check updateCount  to make sure its what am thinking